pkg/activator/net/revision_backends.go

```diff
@@ -119,12 +107,94 @@ type destHealth struct {
 	health bool
 }
 
+func (rw *revisionWatcher) hasHealthyClusterIP() bool {
+	return rw.healthyClusterDest != ""
+}
+
+func (rw *revisionWatcher) revisionReady() bool {
+	return len(rw.dests) > 0
+}
+
+func (rw *revisionWatcher) getK8sPrivateService() (*corev1.Service, error) {
+	selector, err := labels.Parse(fmt.Sprintf("%s=%s,%s=%s",
+		serving.RevisionLabelKey, rw.rev.Name,
+		networking.ServiceTypeKey, networking.ServiceTypePrivate))
+	if err != nil {
+		return nil, err
+	}
+	svcList, err := rw.serviceLister.Services(rw.rev.Namespace).List(selector)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(svcList) == 0 {
+		return nil, fmt.Errorf("Found no private service for revisoin %v", rw.rev)
+	}
+
+	if len(svcList) > 1 {
```

Comment:
could group both to 

'Expected exactly 1 private service, found %d'

when `len(svcList) != 1`

Reply:
fixed with switch